### DISCUSSION TITLE
Parameters for image generation

### DISCUSSION THREAD

**Discussion by daniij** [xtekky/gpt4free#2509] [Q&A]
What parameters can I use to generate images other than model, prompt or response format? For example I tried to use the aspect_ratio parameter, but I didn't notice any change.

**Comment by alexkuu** [comment]
Really interested in this topic as well. Tried aspect_ratio in different formats for all models -it is not doing anything

**Reply by daniij** [reply]
I've learnt on my own that if you use the PollinationsAI provider, you can send the width and height of the photo, but not all values are suitable (you have to experiment). You can also send enhancement parameter, which will improve your prompt. All images that are generated with the help of this provider get to him in the gallery on the site. The private flag will remove the promtp for your image from the site, but the image on the site will remain.
```
async def main():
    client = AsyncClient(image_provider=PollinationsAI)
    
    response = await client.images.generate(
        prompt="photo of dog",
        model="flux",
        response_format="url",
        enhance=True,
        private=True,
        width=2304,
        height=1024
    )
    
    image_url = response.data[0].url
    print(f"Generated image URL: {image_url}")

asyncio.run(main())
```
There are more parameters you can see them by examining the url of the link. (Not what is found by clicking on it): https://image.pollinations.ai/prompt/photo%20of%20dog?seed=7398&width=2304&height=1024&model=flux&nologo=True&private=True&enhance=True&safe=False

**Reply by hlohaus** [reply]
To @alexkuu and @daniij 

All providers support the following:

`print(g4f.Provider.DeepInfra.params)`

and

`if "width" in g4f.Provider.DeepInfra.get_parameters(): print(True)`

Is this helpful?
